packages/expo-constants/src/ExponentConstants.web.ts

```diff
@@ -1,27 +1,42 @@
-import uuidv4 from 'uuid/v4';
 import UAParser from 'ua-parser-js';
+import uuidv4 from 'uuid/v4';
 
 const ExpoPackageJson = require('expo/package.json');
 
+const parser = new UAParser();
+const ID_KEY = 'EXPO_CONSTANTS_INSTALLATION_ID';
+
+declare var process: any;
+declare var navigator: Navigator;
+declare var location: Location;
+declare var localStorage: Storage;
+
+const _sessionId = uuidv4();
+
 export default {
-  _sessionId: uuidv4(),
+  get name(): string {
+    return 'ExponentConstants';
+  },
   get appOwnership() {
     return 'expo';
   },
-  get deviceId() {
-    console.warn(`ExponentConstants.deviceId: is unimplemented on this platform.`);
-    return null;
-  },
-  get name(): string {
-    return 'ExponentConstants';
+  get installationId() {
+    let installationId = localStorage.getItem(ID_KEY);
```

Comment:
Local storage sometimes can fail in browsers (e.g. private mode):

```suggestion
  try {
    let installationId = localStorage.getItem(ID_KEY);
  } catch (e) {
    return _sessionId; // or some other id that is stable for the page's lifetime but doesn't rely on local storage
  }
```